Collision/src/main/java/dk/gruppeseks/bodtrd/collision/CollisionHandler.java

```diff
@@ -0,0 +1,331 @@
+        double collidedy = colPos.getY();
+        double othery = other.get(Position.class).getY();
+        // http://ericleong.me/research/circle-circle/ Need this link for bullet bounce or similar.
+        Vector2 distanceBetweenObjects = new Vector2(collidedx - otherx, collidedy - othery);
+        distanceBetweenObjects.setMagnitude(colBody.getHeight() / 2 + other.get(Body.class).getHeight() / 2);
+        return new Position(otherx + distanceBetweenObjects.getX(), othery + distanceBetweenObjects.getY());
+    }
+
+    /**
+     * Provides a proper response where all kinetic energy is lost for rectangle circle collision, where the circle is the object acting on
+     * collision.
+     *
+     * @param collided
+     * The object that will be acting to the collision.
+     * @param other
+     * The object the acting object collided with.
+     * @return Returns the new suggested position of the collided object.
+     */
+    public static Position collisionResponseCircleRectangle(Position circPos, Body circBody, Position rectPos, Body rectBody) // HASHTAG FUCKING CHECKED
+    {
+        double collidedx = circPos.getX();
+        double collidedy = circPos.getY();
+        double otherx = rectPos.getX();
+        double othery = rectPos.getY();
+        double collidedHeight = circBody.getHeight();
+        double collidedWidth = circBody.getWidth();
+        double otherHeight = rectBody.getHeight();
+        double otherWidth = rectBody.getWidth();
+        double collidedCenterX = getCenter(circPos, circBody).getX();
+        double collidedCenterY = getCenter(circPos, circBody).getY();
+        double otherCenterX = getCenter(rectPos, rectBody).getX();
+        double otherCenterY = getCenter(rectPos, rectBody).getY();
+
+        // If ball is to the right of other and if its center is within the
+        if (collidedCenterX > otherx && collidedCenterX < otherx + otherWidth)
+        {
+            if (collidedCenterY < otherCenterY)
+            {
+                return new Position(collidedx, othery - collidedHeight);
+            }
+            else
+            {
+                return new Position(collidedx, othery + otherHeight);
+            }
+
+        }
+        else if (collidedCenterY > othery && collidedCenterY < othery + otherHeight)
+        {
+            if (collidedCenterX < otherCenterX)
+            {
+                return new Position(otherx - collidedWidth, collidedy);
+            }
+            else
+            {
+                return new Position(otherx + otherWidth, collidedy);
+            }
+        }
+
+        double cornerX = otherx;
+        double cornerY = othery;
+        if (collidedCenterX > otherx)
+        {
+            cornerX += otherWidth;
+            if (collidedCenterY > othery)
+            {
+                cornerY += otherHeight;
+            }
+
+        }
+        else if (collidedCenterX < otherx)
+        {
+            if (collidedCenterY > othery)
```

Comment:
so is this meant to be true for cases where collidedCenterY > othery && collidedCenterX != otherx. or should it be totally independent of collidedCenterX? this code is pretty confusing